app/controllers/api/api_export_controller.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+# API::ApiExportController: for exporting subjects via the API
+class Api::ApiExportController < ApplicationApiController
+  before_action do
+    doorkeeper_authorize!(
+      :'system/Patient.read',
+      :'system/Patient.*'
+    )
+    set_client_app
+  end
+
+  before_action only: %i[nbs_patients] do
+    status_not_acceptable unless request.headers['Accept']&.include?('application/zip')
+  end
+
+  def set_client_app
+    @current_client_app = OauthApplication.find_by(id: doorkeeper_token&.application_id)
+    status_unauthorized if @current_client_app.nil?
+  end
+
+  # Multi patient PHDC export
+  def nbs_patients
+    search_params = params.slice('workflow', 'monitoring', 'caseStatus', 'updatedAt')
+    patients = search_params.blank? ? Patient.none : Jurisdiction.find_by(id: @current_client_app[:jurisdiction_id])&.all_patients_excluding_purged
```

Comment:
No need to do the searching yourself here, let Rails do the work:
```
search_params.blank? ? Patient.none : @current_client_app.jurisdiction&.all_patients_excluding_purged
```

Reply:
Gothca, cleaned up in  49a927a.